### DISCUSSION TITLE
IP import into Vivado?

### DISCUSSION THREAD

**Discussion by DrBwts** [fastmachinelearning/hls4ml#1108] [Q&A]
I have an IP that was generated from the tutorials example `model_1` by using `hls_model.build(csim=False, export=True)`

I targeted the xc7a100tcsg324-1 device on compilation (the device on the Nexys A7 100T board) & exported the IP files. No problems with compilation & the `ip` folder exists in the `imp` folder

I created a new Vivado 2024.1 RTL project targeting the Nexys board but for the life of me I cant work out how to import the IP.

I have trawled through the AMD docs but can only find how to create an IP from with in Vivado or import an IP from the catalogue. The closest I've got in the Manage IP menu which just closes my project & shows the already existent IP with no option to add my own.

Any help greatly appreciated.

**Comment by norahatem** [comment]
Hello,

I’m also using the Nexys A7 (xc7a100t-csg324-1) and have successfully exported the ip and imported it into Vivado (I guess it was you who posted the same question on the Digilent forms, so thank you, that was super helpful!).

I was wondering what your next step was after importing the IP? I’m trying to deploy and validate the model on hardware, but unlike the tutorials (which use a pynq board), the Nexys A7 does not have a hard processor. Did you end up using a MicroBlaze softcore as an AXI master, or did you take a different approach to handle data transfer and testing?

I would prefer not to go down the MicroBlaze route unless necessary, but I’m unsure what the cleanest way is to integrate and validate the IP on this board.

Any insight would be greatly appreciated. Thanks!

Nora
